cli/util/cancel.go

```diff
@@ -0,0 +1,68 @@
+package util
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/nspcc-dev/neo-go/cli/options"
+	"github.com/nspcc-dev/neo-go/cli/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/actor"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/urfave/cli"
+)
+
+func cancelTx(ctx *cli.Context) error {
+	args := ctx.Args()
+	if len(args) == 0 {
+		return cli.NewExitError("Transaction hash is missing", 1)
+	} else if len(args) > 1 {
+		return cli.NewExitError("only one transaction hash is accepted", 1)
+	}
+
+	txHash, err := util.Uint256DecodeStringLE(strings.TrimPrefix(args[0], "0x"))
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Invalid tx hash: %s", args[0]), 1)
+	}
+
+	gctx, cancel := options.GetTimeoutContext(ctx)
+	defer cancel()
+
+	c, err := options.GetRPCClient(gctx, ctx)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to create RPC client: %w", err), 1)
+	}
+
+	txOut, err := c.GetRawTransactionVerbose(txHash)
+	if !txOut.Blockhash.Equals(util.Uint256{}) || err == nil {
+		return cli.NewExitError(fmt.Errorf("transaction %s is already in the blockchain", txHash), 1)
+	}
+	acc, w, err := smartcontract.GetAccFromContext(ctx)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	defer w.Close()
+
+	a, err := actor.NewSimple(c, acc)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	script := byte(opcode.RET)
+
+	resHash, _, err := a.SendTunedRun([]byte{script}, []transaction.Attribute{{Type: transaction.ConflictsT, Value: &transaction.Conflicts{Hash: txHash}}}, func(r *result.Invoke, t *transaction.Transaction) error {
+		err := actor.DefaultCheckerModifier(r, t)
+		if err != nil {
+			return err
+		}
+		t.NetworkFee = int64(math.Max(float64(t.NetworkFee), float64(txOut.NetworkFee)+1))
```

Comment:
Please, don't use floats for money-like objects, we need precise calculations. Compare them manually. The idea itself is correct.